File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.java
import java.util.*;
class Solution {
    public int solution(int n, int w, int num) {
        int answer = 0;
        int h = (n-1) / w + 1;
        int[][] array = new int[h][w];
        int number = 1;
        int numR = 0;
        int numC = 0;
        for(int i=1; i<=h; i++){
            int r = h-i;
            if(i % 2 == 1){     // 정방향
                for(int c=0; c<w; c++){
                    array[r][c] = number;
                    if(number == num){
                        numR = r;
                        numC = c;
                    }
                    number++;
                }
            } else {            // 역방향
                for(int c=w-1; c>=0; c--){
                    array[r][c] = number;
                    if(number == num){
                        numR = r;
                        numC = c;
                    }
                    number++;
                }
            }
        }
        for(int r=0; r<=numR; r++){
            if(array[r][numC] > n){
                continue;
            }
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    answer = 0
    h = (n-1) // w + 1
    array = [[0] * w for c in range(h)]
    number = 1
    for i in range(1 , h+1):
        r = h - i
        if i % 2 == 1:  # 정방향
            for c in range(w):
                array[r][c] = number
                if number == num:
                    numR = r
                    numC = c
                number += 1
        else :          # 역방향
            for c in range(w-1, -1, -1):
                array[r][c] = number
                if number == num:
                    numR = r
                    numC = c
                number += 1
    for r in range(numR+1):
        if array[r][numC] > n:
            continue;
        answer += 1
    return answer